haystack/utils/deepsetcloud.py

```diff
@@ -325,6 +378,24 @@ def list_pipeline_configs(self, workspace: Optional[str] = None, headers: dict =
         generator = self.client.get_with_auto_paging(url=pipelines_url, headers=headers)
         return generator
 
+    def save_pipeline_config(
+        self, config: dict, pipeline_config_name: str, workspace: Optional[str] = None, headers: dict = None
+    ):
+        config["name"] = pipeline_config_name
```

Comment:
What if `config["name"]` already has a value? I'd do a small validation here.

(same for `update_pipeline_config`)

Reply:
DC actually takes the desired pipeline_config_name from here. So we always want to overwrite it at this point.